src/main/java/com/nirmata/workflow/details/WorkflowManagerKafkaImpl.java

```diff
@@ -498,4 +506,47 @@ private List<QueueConsumer> makeTaskConsumers(QueueFactory queueFactory, List<Ta
 
         return builder.build();
     }
+
+    @Override
+    public void submitRootTaskDirect(Task task) {
+        try {
+            RunId runId = RunId.newRandomIdWithPrefix(DIRECT_SUBMIT_RUN_ID_PREFIX);
+
+            StartedTask startedTask = new StartedTask(this.getInstanceName(),
+                    LocalDateTime.now(Clock.systemUTC()), 0);
+            storageMgr.setStartedTask(runId, task.getTaskId(), this.getSerializer().serialize(startedTask));
```

Comment:
No storage manager here. There could be numerous directly submitted tasks with no relation between them.